### PR TITLE
COS/GCE: bump the max pids for the docker service

### LINKED ISSUE #51977: COS m60 set low pids limit for docker.
Forked from https://github.com/kubernetes/kubernetes/issues/51903#issuecomment-327290332.

Since m60, COS set 4000 pids limit for docker.service.

However, currently all containerd-shims are also inside the cgroup, and they are per-container.
Each containerd-shim has more than 10 threads based on the node size:
```
$ $ ps aux | grep containerd-shim
root       4218  0.0  0.0 293100 12588 ?        Ssl  Aug10   1:42 docker-containerd -l unix:///var/run/docker/libcontainerd/docker-containerd.sock --shim docker-containerd-shim --metrics-interval=0 --start-timeout 2m --state-dir /var/run/docker/libcontainerd/containerd --runtime docker-runc
root      10240  0.2  0.0 422176  5476 ?        Sl   13:12   0:00 docker-containerd-shim 04065c55d068762a3b55d277874ddb4897315f113f1ac865975ae0ca7eb78151 /var/run/docker/libcontainerd/04065c55d068762a3b55d277874ddb4897315f113f1ac865975ae0ca7eb78151 docker-runc
lantaol   10328  0.0  0.0  23780  2228 pts/21   S+   13:12   0:00 grep --color=auto containerd-shim

$ pstree 10240
docker-containe─┬─nginx───nginx
                └─10*[{docker-containe}]
```

It's not hard to hit the limit, say 100 3-container pods may hit the limit.

We may want to:
1) Either patch kubernetes to change the limit to higher value;
2) Or patch COS directly, since this may be a regression for other COS user, too.

@yujuhong @dchen1107 @adityakali @wonderfly 
/cc @kubernetes/sig-node-bugs

### PR DESCRIPTION
**Pull request by yujuhong**
**What this PR does / why we need it**:
TasksMax limits how many threads/processes docker can create. Insufficient limit affects container starts.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*:
fixes #51977

**Special notes for your reviewer**:

**Release note**:
```release-note
In GCE with COS, increase TasksMax for Docker service to raise cap on number of threads/processes used by containers.
```

### CODE REVIEW

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -811,20 +811,26 @@ function assemble-docker-flags {
...
+  fi
 
-    systemctl daemon-reload
+  # COS m60 sets the default TasksMax that is too low for us. Bump it up.
+  # (https://github.com/kubernetes/kubernetes/issues/51977)
+  echo "Extend the docker.service configuration to set a higher pids limit"
+  mkdir -p /etc/systemd/system/docker.service.d
+  cat <<EOF >/etc/systemd/system/docker.service.d/02tasksmax.conf
+[Service]
+TasksMax=10000
```
Comment (dchen1107) [inline cluster/gce/gci/configure-helper.sh]: Can we make this even bigger, something like 1048600?

**Comment by dchen1107** [comment]
@yujuhong We might have to cherrypick this back to 1.6 and 1.7.

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (yujuhong) [reply cluster/gce/gci/configure-helper.sh]: The current number 4915 is the "15% of "max number of tasks on the system" according to the COS team. How about setting it to `infinity` (i.e., no limit)?

**Comment by yujuhong** [comment]
Updated the PR.

**Comment by Random-Liu** [comment]
LGTM

**Comment by wojtek-t** [comment]
Cherrypick for 1.7 approved.

**Pull request merged**